homeassistant/components/hue/config_flow.py

```diff
@@ -194,26 +195,30 @@ async def async_step_ssdp(self, discovery_info: ssdp.SsdpServiceInfo) -> FlowRes
         """
         # Filter out non-Hue bridges #1
         if (
-            discovery_info.get(ssdp.ATTR_UPNP_MANUFACTURER_URL)
+            discovery_info.upnp.get(ssdp.ATTR_UPNP_MANUFACTURER_URL)
             not in HUE_MANUFACTURERURL
         ):
             return self.async_abort(reason="not_hue_bridge")
 
         # Filter out non-Hue bridges #2
         if any(
-            name in discovery_info.get(ssdp.ATTR_UPNP_FRIENDLY_NAME, "")
+            name in discovery_info.upnp.get(ssdp.ATTR_UPNP_FRIENDLY_NAME, "")
             for name in HUE_IGNORED_BRIDGE_NAMES
         ):
             return self.async_abort(reason="not_hue_bridge")
 
         if (
-            not discovery_info.get(ssdp.ATTR_SSDP_LOCATION)
-            or ssdp.ATTR_UPNP_SERIAL not in discovery_info
+            not discovery_info.ssdp_location
+            or ssdp.ATTR_UPNP_SERIAL not in discovery_info.upnp
         ):
             return self.async_abort(reason="not_hue_bridge")
 
-        host = urlparse(discovery_info[ssdp.ATTR_SSDP_LOCATION]).hostname
-        bridge = await self._get_bridge(host, discovery_info[ssdp.ATTR_UPNP_SERIAL])
+        host = urlparse(discovery_info.ssdp_location).hostname
+        if TYPE_CHECKING:
+            assert host is not None
```

Comment:
can you enlighten me why this magic is in place ?

Reply:
`urlparse().hostname` returns `str | None` but `_get_bridge()` doesn't accept `None` and it is causing an issue for `mypy`.

Reply:
Honestly, mypy is complaining correctly... we should not silence it, we should act on it.

Reply:
this looks a bit strange to me... Either tell mypy to ignore this line or abort on a None value instead.
I prefer to abort because none of the following code can handle the host to be None

Reply:
I have adjusted the check and the conversion to this:
```python
    host = str(urlparse(discovery_info.ssdp_location).hostname)
    if not host:
            return self.async_abort(reason="not_hue_bridge")
```

Reply:
looks good now with your adjustment!

Reply:
We do need a test for it though, in order to hit this line, as we require config flows to be 100% covered.

Reply:
unresolved.

Reply:
resolved.

Reply:
See my suggestions. The hostname property can be either None or a string.
So that's what the code should act on. Do not "convert" the value to string to make tests pass but adjust the logic.
There might as well be a small bug in here for a long time but in that case we should fix it, not work around it.

so:
- abort when hostname is None
- make sure the test covers that receiving None on hostname attribute aborts the config flow.